### PR TITLE
Use `StringInput` for all text inputs

### PR DESCRIPTION
**Pull request by distantnative**
### Refactored
- use `k-string-input` for all text inputs

### CODE REVIEW

File: panel/src/components/Forms/Input/PasswordInput.vue [hunk 1]
```diff
@@ -19,7 +25,6 @@ export const props = {
  * @example <k-input :value="password" @input="password = $event" name="password" type="password" />
  */
 export default {
-	extends: TextInput,
-	mixins: [props]
```
Comment (bastianallgeier) [inline panel/src/components/Forms/Input/PasswordInput.vue:23]: Why are the props removed here?

File: panel/src/components/Forms/Input/SlugInput.vue [hunk 2]
```diff
@@ -17,7 +17,7 @@
 		spellcheck="false"
 		type="text"
 		class="k-text-input"
-		v-on="listeners"
+		@input="$emit('input', $event)"
```
Comment (bastianallgeier) [inline panel/src/components/Forms/Input/SlugInput.vue:20]: Not sure about this one. Wouldn't this remove the validate listener from the text input, which could be a breaking change?

File: panel/src/components/Forms/Input/TelInput.vue [hunk 3]
```diff
@@ -20,7 +27,6 @@ export const props = {
  * @example <k-input :value="tel" @input="tel = $event" name="tel" type="tel" />
  */
 export default {
-	extends: TextInput,
-	mixins: [props]
```
Comment (bastianallgeier) [inline panel/src/components/Forms/Input/TelInput.vue:24]: See above.

File: panel/src/components/Forms/Input/UrlInput.vue [hunk 4]
```diff
@@ -23,7 +29,6 @@ export const props = {
  * @example <k-input :value="url" @input="url = $event" name="url" type="url" />
  */
 export default {
-	extends: TextInput,
-	mixins: [props]
```
Comment (bastianallgeier) [inline panel/src/components/Forms/Input/UrlInput.vue:27]: One more

**Review by bastianallgeier — COMMENTED** [review]
I think this clean-up is really good. But I wonder if we are again running into breaking change territory with the CSS classes. So far, all the refactored fields hat a k-text-input class which is now replaced by k-string-input. This could lead to problems.

File: panel/src/components/Forms/Input/SlugInput.vue (hunk 2 above)
Comment (distantnative) [reply panel/src/components/Forms/Input/SlugInput.vue:20]: True probably needs `@invalid="($invalid, $v) => $emit('invalid', $invalid, $v)"` as the other inputs as well

Comment (distantnative) [reply panel/src/components/Forms/Input/SlugInput.vue:20]: Oh not, cause it's a native `input` element, I don't think the previous invalid event listener had any real function on this native element

**Comment by distantnative** [comment]
@bastianallgeier that's true. maybe rather `v5/develop` then?

**Review by bastianallgeier — APPROVED** [review]

**Comment by bastianallgeier** [comment]
One last thought before we move on with this. Can we still properly work on our inputs during v4 when we merge this and also maybe the vuelidate stuff or do we need to do more in v4 as preparation? I just don't want us to get in the way with ourselves.

**Comment by distantnative** [comment]
I think interior be good to have this before the vuelidate work (it's basically already in the vuelidate PR but thought it should be separate).

Of course we'll need to be careful about merge conflicts if we make changes to those specific inputs in v4.x

**Comment by bastianallgeier** [comment]
Ok, lets just go for it. It served us well during the v4 alphas to move ahead with such tasks and then just keep an eye on merging changes back in quickly.

**Pull request merged**